org.eclipse.gef.examples.shapes/src/org/eclipse/gef/examples/shapes/action/ColorPaletteAction.java

```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Patrick Ziegler and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Patrick Ziegler - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.gef.examples.shapes.action;
+
+import org.eclipse.ui.IEditorPart;
+
+import org.eclipse.gef.GraphicalViewer;
+import org.eclipse.gef.ui.actions.EditorPartAction;
+import org.eclipse.gef.ui.palette.PaletteViewer;
+
+import org.eclipse.gef.examples.shapes.palette.ShapesColorPalette;
+
+/**
+ * Flips between the custom and the default color palette whenever executed.
+ */
+public class ColorPaletteAction extends EditorPartAction {
```

Comment:
shouldn't we use for new code the command, handler, menu extension points?

Reply:
You're probably right... But then I would go all the way and use the E4 mechanism. As much as I hate dealing with the e4xmi files...

This is how it looks like in the IDE:
![image](https://github.com/user-attachments/assets/22b1981f-a362-46fe-aef6-c91c22b00726)

Reply:
Though it's probably not wrong to also have an E4 example...